test/realtime_box_tests.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest.h>
+#include <realtime_tools/realtime_box.h>
+
+using realtime_tools::RealtimeBox;
+
+class DefaultConstructable
+{
+  public:
+    DefaultConstructable() : number_(42) {};
+    ~DefaultConstructable() {};
+    int number_;
+};
+
+TEST(RealtimeBox, default_construct)
+{
+  DefaultConstructable thing;
+  thing.number_ = 5;
+
+  RealtimeBox<DefaultConstructable> box;
+  box.get(thing);
+
+  EXPECT_EQ(42, thing.number_);
+}
+
+TEST(RealtimeBox, initial_value)
+{
+  RealtimeBox<double> box(3.14);
+  double num = 0.0;
+  box.get(num);
+  EXPECT_DOUBLE_EQ(3.14, num);
+}
+
+TEST(RealtimeBox, set_and_get)
+{
+  RealtimeBox<char> box('a');
+
+  {
+    const char input = 'z';
+    box.set(input);
+  }
+
+  char output = 'a';
+  box.get(output);
+  EXPECT_EQ('z', output);
+}
+
+int main(int argc, char ** argv) {
```

Comment:
This standard main could be removed and replaced by linking to `GMOCK_MAIN_LIBRARIES` in the cmake. I think it's always better when we have less C++ to maintain, no matter how simple.

Reply:
Of course this doesn't apply in cases where you have to call `ros::init`.

Reply:
Uses `GMOCK_MAIN_LIBRARIES` for box and buffer tests in ac9bf90f27c2c8464c9c7cf2775960ca1ba3d95f